src/app/components/harvest/screens/metadata-review/metadata-review.component.ts

```diff
@@ -248,22 +248,29 @@ export class MetadataReviewComponent
       : row.harvestItem?.path;
 
   public updateHarvestWithMappingChange(): void {
-    const mappings: HarvestMapping[] = [];
-
     /*
-     * Iterate through all the rows, extract folders which have mappings and
-     * push that list to the mappings array. We could use a filter and map for
-     * this, however it's likely less performant
-     */
+    * Iterate through all the rows, extract folders which have mappings and
+    * push that list to the mappings array. We could use a filter and map for
+    * this, however it's likely less performant
+    */
     this.rows.forEach((row): void => {
       if (row.rowType === RowType.folder && isInstantiated(row.mapping)) {
-        mappings.push(row.mapping);
+
+        if (this.harvest.mappings.find(mapping => mapping.path === row.mapping.path)) {
+          // since the path attribute is a unique identifier on a mapping, we can safely assume
+          // that there will ever only be one mapping returned from the bellow filter
+          Object.assign(this.harvest.mappings.filter(mapping => mapping.path === row.mapping.path)[0], row.mapping);
```

Comment:
two passes are redundant here, find the object first, and then update it by reference second. Don't need to do `.filter` again